cranelift-reader/src/parser.rs

```diff
@@ -770,7 +807,11 @@ impl<'a> Parser<'a> {
     pub fn parse_test_commands(&mut self) -> Vec<TestCommand<'a>> {
         let mut list = Vec::new();
         while self.token() == Some(Token::Identifier("test")) {
-            list.push(TestCommand::new(self.consume_line()));
+            let line = self.consume_line();
+            if line.contains("run") {
+                self.default_calling_convention = CallConv::triple_default(&Triple::host()); // use the host's default calling convention
```

Comment:
Looking at how it's used in parse_test, I think it would badly interfere if some passes are given to the command-line. Instead, could we set the calling convention in parse_test after the `match passes`, if there's `any` run command in the list of commands?